enclave-runtime/src/top_pool_execution.rs

```diff
@@ -92,25 +99,54 @@ fn execute_top_pool_trusted_calls_internal() -> Result<()> {
 
 	let slot_beginning_timestamp = duration_now();
 
-	let parentchain_import_dispatcher = get_triggered_dispatcher_from_solo_or_parachain()?;
-
-	let validator_access = get_validator_accessor_from_solo_or_parachain()?;
+	let integritee_parentchain_import_dispatcher =
+		get_triggered_dispatcher_from_integritee_solo_or_parachain()?;
+	let maybe_target_a_parentchain_import_dispatcher =
+		get_triggered_dispatcher_from_target_a_solo_or_parachain().ok();
+	let maybe_target_b_parentchain_import_dispatcher =
+		get_triggered_dispatcher_from_target_b_solo_or_parachain().ok();
+
+	// todo: is this really correct? are these blocks already processed into top-pool?
```

Comment:
This seems weird to me too actually.

Reply:
the whole thing would not work if this wasn't true, so I removed the comment which I wrote before it worked ;-)